components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/MobileNumberVerificationHandler.java

```diff
@@ -0,0 +1,380 @@
+
+        if (IdentityEventConstants.Event.POST_SET_USER_CLAIMS.equals(eventName)) {
+            postSetUserClaimOnMobileNumberUpdate(user, userStoreManager);
+        }
+    }
+
+    @Override
+    public void init(InitConfig configuration) throws IdentityRuntimeException {
+
+        super.init(configuration);
+    }
+
+    @Override
+    public int getPriority(MessageContext messageContext) {
+
+        return 50;
+    }
+
+    /**
+     * Store verification details in the recovery data store and initiate notification.
+     *
+     * @param user  User.
+     * @param verificationPendingMobileNumber Updated mobile number that is pending verification.
+     * @throws IdentityEventException
+     */
+    private void initNotificationForMobileNumberVerificationOnUpdate(User user, String verificationPendingMobileNumber)
+            throws IdentityEventException {
+
+        UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
+
+        try {
+            userRecoveryDataStore.invalidate(user, RecoveryScenarios.MOBILE_VERIFICATION_ON_UPDATE,
+                    RecoverySteps.VERIFY_MOBILE_NUMBER);
+            String secretKey = Utils.generateSecretKey(NotificationChannels.SMS_CHANNEL.getChannelType(),
+                    user.getTenantDomain(), String.valueOf(RecoveryScenarios.MOBILE_VERIFICATION_ON_UPDATE));
+            UserRecoveryData recoveryDataDO = new UserRecoveryData(user, secretKey,
+                    RecoveryScenarios.MOBILE_VERIFICATION_ON_UPDATE, RecoverySteps.VERIFY_MOBILE_NUMBER);
+            /* Mobile number is persisted in remaining set ids to maintain context information about the mobile number
+            associated with the verification code generated. */
+            recoveryDataDO.setRemainingSetIds(verificationPendingMobileNumber);
+            userRecoveryDataStore.store(recoveryDataDO);
+            triggerNotification(user, secretKey, Utils.getArbitraryProperties(), verificationPendingMobileNumber);
+        } catch (IdentityRecoveryException e) {
+            throw new IdentityEventException("Error while sending notification to user: " +
+                    user.toFullQualifiedUsername() + " for mobile verification on update.", e);
+        }
+    }
+
+    /**
+     * Trigger the SMS notification.
+     *
+     * @param user      User.
+     * @param code      SMS OTP.
+     * @param props     Other properties.
+     * @param verificationPendingMobileNumber Mobile number to which the SMS should be sent.
+     * @throws IdentityRecoveryException
+     */
+    private void triggerNotification(User user, String code, Property[] props, String
+            verificationPendingMobileNumber) throws IdentityRecoveryException {
+
+        String notificationType = IdentityRecoveryConstants.NOTIFICATION_TYPE_VERIFY_MOBILE_ON_UPDATE;
+
+        if (log.isDebugEnabled()) {
+            log.debug("Sending : " + notificationType + " notification to user : " + user.toFullQualifiedUsername());
```

Comment:
```suggestion
            log.debug("Sending: " + notificationType + " notification to user: " + user.toFullQualifiedUsername());
```

Reply:
Addressed in https://github.com/wso2-extensions/identity-governance/pull/414/commits/1f1503b13f90b814eca821c6fca73b9af4c61052